docs/source/reference/api-server/api-server-tunning.rst

```diff
@@ -0,0 +1,152 @@
+.. _sky-api-server-performance-best-practices:
+
+SkyPilot API Server Performance Best Practices
+==============================================
+
+This page describes performance best practices for remote SkyPilot API server in :ref:`team deployment <sky-api-server>`.
+
+Tuning API server resources
+---------------------------
+
+The number of requests that the API server can handle concurrently is proportional to the resources (CPU cores and memory) allocated to it. Internally, requests are categorized into two different types and handled in a first-in-first-out manner:
+
+* ``Long-running request``: request that takes long time and more resources to run, including ``launch``, ``exec``, ``jobs.launch``, etc.
+* ``Short-running request``: request that takes short time or less resources to run, including ``status``, ``logs``, etc.
+
+.. note::
+
+    Though a task (or job) can run for any length of time, concurrent tasks does not occupy the concurrency. Because once a task is submitted to the cluster, it will be detached and no longer takes any resources off the API server.
```

Comment:
I think we should rephrase this somewhat, and pull it out of note box. How about something like:

Requests are queued and processed by the API server. Therefore, they only take resources off the API server when they are in queue or being processed. Once requests are processed and remote clusters start doing real work, they no longer require API server's resources or count against its concurrency limit.

For example, long-running requests for ``launch`` and ``exec`` no longer take resources off the API server once a cluster has been provisioned, or once a job has been submitted to a cluster, respectively.

Reply:
Please check accuracy.

Reply:
Exactly! Thanks for the refinement @concretevitamin !

Reply:
Sounds good, please go ahead.